test/framework/managedseedframework.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	CBeforeEach(func(ctx context.Context) {
+		f.CommonFramework.BeforeEach()
+		f.GardenerFramework.BeforeEach()
+		f.BeforeEach(ctx)
+	}, 8*time.Minute)
+	CAfterEach(f.AfterEach, 10*time.Minute)
+
+	return f
+}
+
+// BeforeEach should be called in ginkgo's BeforeEach.
+// It merges and validates the managed seed framework configuration.
+func (f *ManagedSeedFramework) BeforeEach(ctx context.Context) {
+	f.Config = mergeManagedSeedConfig(f.Config, managedSeedConfig)
+	validateManagedSeedConfig(f.Config)
+}
+
+// AfterEach should be called in ginkgo's AfterEach.
+// It dumps the managed seed framework state if the test failed.
+func (f *ManagedSeedFramework) AfterEach(ctx context.Context) {
+	if ginkgo.CurrentGinkgoTestDescription().Failed {
+		f.DumpState(ctx)
+	}
+}
+
+func validateManagedSeedConfig(cfg *ManagedSeedConfig) {
+	if cfg == nil {
+		ginkgo.Fail("no configuration provided")
+	}
+	if !StringSet(cfg.ManagedSeedName) {
+		ginkgo.Fail("You should specify a name for the managed seed")
+	}
+	if !StringSet(cfg.ShootName) {
+		ginkgo.Fail("You should specify the name of the shoot to be registered as seed")
+	}
+}
+
+func mergeManagedSeedConfig(base, overwrite *ManagedSeedConfig) *ManagedSeedConfig {
+	if base == nil {
+		return overwrite
+	}
+	if overwrite == nil {
+		return base
+	}
+	if overwrite.GardenerConfig != nil {
+		base.GardenerConfig = overwrite.GardenerConfig
+	}
+	if StringSet(overwrite.ManagedSeedName) {
+		base.ManagedSeedName = overwrite.ManagedSeedName
+	}
+	if StringSet(overwrite.ShootName) {
+		base.ShootName = overwrite.ShootName
+	}
+	if overwrite.DeployGardenlet {
+		base.DeployGardenlet = overwrite.DeployGardenlet
+	}
+	if StringSet(overwrite.BackupProvider) {
+		base.BackupProvider = overwrite.BackupProvider
+	}
+	return base
+}
+
+// RegisterManagedSeedFrameworkFlags adds all flags that are needed to configure a managed seed framework.
+func RegisterManagedSeedFrameworkFlags() *ManagedSeedConfig {
+	_ = RegisterGardenerFrameworkFlags()
+
+	newCfg := &ManagedSeedConfig{}
+
+	flag.StringVar(&newCfg.ManagedSeedName, "managed-seed-name", "", "name of the managed seed")
+	flag.StringVar(&newCfg.ShootName, "shoot-name", "", "name of the shoot to be registered as seed")
+	flag.BoolVar(&newCfg.DeployGardenlet, "deploy-gardenlet", true, "indicates if gardenlet should be deployed to the shoot, default is true")
+	flag.StringVar(&newCfg.BackupProvider, "backup-provider", "", "seed backup provider")
+
+	managedSeedConfig = newCfg
+
+	return managedSeedConfig
+}
+
+// CreateManagedSeed creates a new managed seed according to the managed seed framework configuration.
+func (f *ManagedSeedFramework) CreateManagedSeed(ctx context.Context) error {
+	if f.GardenClient == nil {
+		return errors.New("no gardener client is defined")
+	}
+
+	// Get shoot
+	shoot := &gardencorev1beta1.Shoot{}
+	if err := f.GardenClient.DirectClient().Get(ctx, kutil.Key(v1beta1constants.GardenNamespace, f.Config.ShootName), shoot); err != nil {
```

Comment:
does the shoot have to reside in the garden namespace?

Reply:
Yes. It has to be in the same namespace as the `ManagedSeed` resource, and `ManagedSeed` resources are currently restricted to the `garden` namespace. In the future this might change if something like "private seeds" is introduced but there are no concrete plans for this.